### PR TITLE
solfuzzer: adds --input-file=FILENAME command line argument (defaulting to stdin)

### PR DESCRIPTION
**Pull request by christianparpart**
While debugging my other PR I had difficulties running solfuzzer on Windows and within VS2017 in debug mode due to the fact, that files are passed via stdin.

Thus I propose this change to allow passing files via command line parameter instead, yet defaulting to old behavior of using stdin if not stated otherwise.

Related to #4924.

### CODE REVIEW

**Comment by chriseth** [comment]
Good idea!
I would propose to use the first unnamed option for the filename, though, just as with `solc`. What do you think?

**Comment by christianparpart** [comment]
I am absolutely fine with that idea. I just didn't know how to realize that in the first place. I'll look into solc's solution and then adapt this one.

**Review by leonardoalt — DISMISSED** [review]
Looks good to me

File: test/tools/fuzzer.cpp [hunk 1]
```diff
@@ -48,15 +49,17 @@ string contains(string const& _haystack, vector<string> const& _needles)
 	return "";
 }
 
-void testConstantOptimizer()
+void testConstantOptimizer(const string& input)
```
Comment (chriseth) [inline test/tools/fuzzer.cpp]: `string const&`

File: test/tools/fuzzer.cpp [hunk 2]
```diff
@@ -167,16 +168,25 @@ Allowed options)",
 			"Run the constant optimizer instead of compiling. "
 			"Expects a binary string of up to 32 bytes on stdin."
 		)
+		(
+			"input-file",
+			po::value<string>(),
+			"Solidity source code input file"
```
Comment (chriseth) [inline test/tools/fuzzer.cpp]: The input is not always a solidity source code.

**Review by chriseth — CHANGES_REQUESTED** [review]

**Review by chriseth — APPROVED** [review]

**Pull request merged**